### PR TITLE
interpret negative values in ReduceAxes ops

### PR DESCRIPTION
**Pull request by szha**

### CODE REVIEW

**Comment by szha** [comment]
@piiswrong

File: src/operator/tensor/broadcast_reduce_op.h [hunk 1]
```diff
@@ -166,44 +168,58 @@ inline TShape ReduceAxesShapeImpl(const TShape& ishape, const TShape& axis,
...
   }
 
-  CHECK_LT(axis[axis.ndim()-1], ishape.ndim())
-    << "Reduction axis " << axis[axis.ndim()-1]
+  TShape axes(axis);
+  for (index_t i = 0; i < axes.ndim(); i++) {
+    if (axes[i] < 0) {
+      axes[i] += ishape.ndim();
+    } else {
+      break;
```
Comment (piiswrong) [inline src/operator/tensor/broadcast_reduce_op.h]: now that axis is not sorted you cannot break here

File: src/operator/tensor/broadcast_reduce_op.h [hunk 2]
```diff
@@ -166,44 +168,58 @@ inline TShape ReduceAxesShapeImpl(const TShape& ishape, const TShape& axis,
...
+    }
+  }
+  std::sort(axes.begin(), axes.end());
+
+  for (index_t i = 1; i < axes.ndim(); i++) {
+    CHECK_LT(axes[i-1], axes[i])
+      << "Reduction axes have duplicates "
+      << axes;
+  }
+  CHECK_LT(axes[axes.ndim()-1], ishape.ndim())
```
Comment (piiswrong) [inline src/operator/tensor/broadcast_reduce_op.h]: also check axes[0] >= 0

Comment (szha) [inline src/operator/tensor/broadcast_reduce_op.h]: added check

**Pull request merged**